From e9e00d035911d3f46f83acebe99d8ead435405a8 Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Mon, 9 Dec 2024 15:33:14 +0100
Subject: [PATCH 36/36] Revert "platform/x86: asus-wmi: Ignore return value
 when writing thermal policy"

This reverts commit aaaf323f6af8ff419a1432738f8261553deba910.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 drivers/platform/x86/asus-wmi.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/drivers/platform/x86/asus-wmi.c b/drivers/platform/x86/asus-wmi.c
index a173f95d6..89f5f4485 100644
--- a/drivers/platform/x86/asus-wmi.c
+++ b/drivers/platform/x86/asus-wmi.c
@@ -3106,6 +3106,7 @@ static int fan_boost_mode_check_present(struct asus_wmi *asus)
 
 static int fan_boost_mode_write(struct asus_wmi *asus)
 {
+	u32 retval;
 	u8 value;
 	int err;
 
@@ -3717,8 +3718,8 @@ static int throttle_thermal_policy_write(struct asus_wmi *asus)
 		value = asus->throttle_thermal_policy_mode;
 	}
 
-	/* Some machines do not return an error code as a result, so we ignore it */
-	err = asus_wmi_set_devstate(asus->throttle_thermal_policy_dev, value, NULL);
+	err = asus_wmi_set_devstate(asus->throttle_thermal_policy_dev,
+				    value, &retval);
 
 	sysfs_notify(&asus->platform_device->dev.kobj, NULL,
 			"throttle_thermal_policy");
@@ -3728,6 +3729,12 @@ static int throttle_thermal_policy_write(struct asus_wmi *asus)
 		return err;
 	}
 
+	if (retval != 1) {
+		pr_warn("Failed to set throttle thermal policy (retval): 0x%x\n",
+			retval);
+		return -EIO;
+	}
+
 	/* Must set to disabled if mode is toggled */
 	if (asus->cpu_fan_curve_available)
 		asus->custom_fan_curves[FAN_CURVE_DEV_CPU].enabled = false;
-- 
2.47.1

